pandas/core/series.py

```diff
@@ -704,6 +706,82 @@ def view(self, dtype=None):
 
     # ----------------------------------------------------------------------
     # NDArray Compat
+    _HANDLED_TYPES = (Index, ExtensionArray, np.ndarray, numbers.Number,
+                      list)  # what other builtins? array? deque? ...
+
+    def __array_ufunc__(
+            self,
+            ufunc: Callable,
+            method: str,
+            *inputs: Any,
+            **kwargs: Any
+    ):
+        # TODO: handle DataFrame
+        from pandas.core.internals.construction import extract_array
+
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not NotImplemented:
+            return result
+
+        # align all the inputs.
+        names = [getattr(x, 'name') for x in inputs if hasattr(x, 'name')]
+        types = tuple(type(x) for x in inputs)
+        # TODO: dataframe
+        alignable = [x for x, t in zip(inputs, types) if issubclass(t, Series)]
+
+        if len(alignable) > 1:
```

Comment:
are we not *always* alignable? when would not be

Reply:
e.g. `np.add(Series, ndarray)` or `np.add(Series, Index)` will not align.